src/Fetcher/fetcher.go

```diff
@@ -0,0 +1,147 @@
+package Fetcher
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+type Repo struct {
+	URL      string
+	Branch   string
+	SpecFile string
+}
+
+func FetchFiles(url, branch, specFile string) error {
```

Comment:
I disagree with this sentiment. It would take extra time to do, doesn't change performance, and in some way obfuscates what the function does since you now have to look at the struct to see what the values are instead of just looking at the arguments

Reply:
If we think this is the way to go (pun intended) then it's fine with me